src/test/java/org/openhab/tools/analysis/checkstyle/test/ManifestPackageVersionCheckTest.java

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.tools.analysis.checkstyle.test;
+
+import java.io.File;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.openhab.tools.analysis.checkstyle.ManifestPackageVersionCheck;
+import org.openhab.tools.analysis.checkstyle.api.AbstractStaticCheckTest;
+
+import com.puppycrawl.tools.checkstyle.DefaultConfiguration;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.utils.CommonUtils;
+
+/**
+ * Tests for {@link ManifestPackageVersionCheck}
+ *
+ * @author Petar Valchev - Initial contribution with tests for imported packages.
+ * @author Aleksandar Kovachev - Added tests for exported packages.
+ *
+ */
+public class ManifestPackageVersionCheckTest extends AbstractStaticCheckTest {
+
+    private static final String TEST_FILE_NAME = "MANIFEST.MF";
+
+    static DefaultConfiguration config;
+
+    @BeforeClass
+    public static void createConfiguration() {
+        config = createCheckConfig(ManifestPackageVersionCheck.class);
+        config.addAttribute("ignoreImportedPackages", "org.apache.*, org.junit.*");
+        config.addAttribute("ignoreExportedPackages", "org.eclipse.smarthome.tool.*");
+    }
+
+    @Override
+    protected DefaultConfiguration createCheckerConfig(Configuration config) {
+        DefaultConfiguration configParent = new DefaultConfiguration("root");
+        configParent.addChild(config);
+        return configParent;
+    }
+
+    @Test
+    public void testImportWithVersion() throws Exception {
+        int lineNumber = 10;
+        String[] expectedMessages = generateExpectedMessages(lineNumber,
+                String.format(ManifestPackageVersionCheck.VERSION_USED_MSG, "javax.xml.bind"));
+        verifyManifest("invalidImportsManifest", expectedMessages);
+    }
+
+    @Test
+    public void testExportWithVersion() throws Exception {
+        int lineNumber = 13;
+        String[] expectedMessages = generateExpectedMessages(lineNumber,
+                String.format(ManifestPackageVersionCheck.VERSION_USED_MSG, "org.eclipse.smarthome.buildtools.other"));
+        verifyManifest("invalidExportsManifest", expectedMessages);
+    }
+
+    @Test
+    public void testMultipleImportsWithVersion() throws Exception {
+        int firstLineNumber = 10;
+        int secondLineNumber = 13;
+        int thirdLineNumber = 14;
+        String[] expectedMessages = generateExpectedMessages(firstLineNumber,
```

Comment:
The readability will improve if you could group the expected line with the exptected message, also static importing the VERSION_USED_MSG and the format method will help in that. 

And as soon as it became readable I would inline the linenumbers because that reads way better:

```java
        String[] expectedMessages = generateExpectedMessages(
            11, format(VERSION_USED_MSG, "com.apache.commons.io"), 
            13, format(VERSION_USED_MSG, "com.apache.commons"),  
            15, format(VERSION_USED_MSG, "com.junit"),
            21, format(VERSION_USED_MSG, "org.eclipse.smarthome.checkstyle")
        );
        verifyManifest("testConfiguratedRegexPackages", expectedMessages);
```